src/Microsoft.Health.Fhir.SqlServer/Features/Search/SearchOptionsExtensions.cs

```diff
@@ -32,5 +32,24 @@ public static SortOrder GetFirstSortOrderForSupportedParam(this SearchOptions se
 
             return sortOrder;
         }
+
+        public static (Core.Models.SearchParameterInfo, SortOrder) GetFirstSupportedSortParam(this SearchOptions searchOptions)
+        {
+            EnsureArg.IsNotNull(searchOptions, nameof(searchOptions));
+
+            foreach (var sortOptions in searchOptions.Sort)
+            {
+                if (sortOptions.searchParameterInfo.IsSortSupported())
+                {
+                    return sortOptions;
+                }
+                else
+                {
+                    throw new SearchParameterNotSupportedException(string.Format(Core.Resources.SearchParameterNotSupported, sortOptions.searchParameterInfo.Name));
```

Comment:
It's not that the search parameter is not supported. It's that we don't support sorting by it.

Reply:
Also, we should tell them which ones are supported for the given resource type.